Source/ACE.Server/Factories/LootGenerationFactory.cs

```diff
@@ -1545,78 +1488,108 @@ private static int GetWorkmanship(int tier)
             return workmanship;
         }
 
-        private static int GetSpellcraft(int spellAmount, int tier)
+        private static int GetSpellcraft(WorldObject wo, int spellAmount, int tier)
         {
-            int spellcraft = 0;
-            switch (tier)
+
+            float minItemSpellCraftRange = 0.0f;
+            float maxItemSpellCraftRange = 0.0f;
+
+            switch (wo.ItemType)
             {
-                case 1:
-                    spellcraft = ThreadSafeRandom.Next(1, 20) + spellAmount * ThreadSafeRandom.Next(1, 4); //1-50
-                    break;
-                case 2:
-                    spellcraft = ThreadSafeRandom.Next(40, 70) + spellAmount * ThreadSafeRandom.Next(1, 5); //40-90
-                    break;
-                case 3:
-                    spellcraft = ThreadSafeRandom.Next(70, 90) + spellAmount * ThreadSafeRandom.Next(1, 6); //80 - 130
-                    break;
-                case 4:
-                    spellcraft = ThreadSafeRandom.Next(100, 120) + spellAmount * ThreadSafeRandom.Next(1, 7); /// 120 - 160
-                    break;
-                case 5:
-                    spellcraft = ThreadSafeRandom.Next(130, 150) + spellAmount * ThreadSafeRandom.Next(1, 8); ///150 - 210
-                    break;
-                case 6:
-                    spellcraft = ThreadSafeRandom.Next(160, 180) + spellAmount * ThreadSafeRandom.Next(1, 9); /// 200-260
-                    break;
-                case 7:
-                    spellcraft = ThreadSafeRandom.Next(230, 260) + spellAmount * ThreadSafeRandom.Next(1, 10); /// 250 - 310
-                    break;
-                case 8:
-                    spellcraft = ThreadSafeRandom.Next(280, 300) + spellAmount * ThreadSafeRandom.Next(1, 11); //300-450
-                    break;
+                case ItemType.MeleeWeapon:
+                case ItemType.Caster:
+                case ItemType.MissileWeapon:
+                case ItemType.Armor:
+                case ItemType.Clothing:
+                case ItemType.Jewelry:
+                    minItemSpellCraftRange = 0.90f;
+                    maxItemSpellCraftRange = 1.05f;
+                    break;
+                case ItemType.Gem:
                 default:
+                    minItemSpellCraftRange = 1.00f;
+                    maxItemSpellCraftRange = 1.00f;
                     break;
             }
 
-            return spellcraft;
+            // Getting the spell difficulty - Maybe a better way to do this.
+            var maxSpellLevel = wo.GetMaxSpellLevel();
+            int maxSpellDiff = 1;
+
+            if (maxSpellLevel == 2)
```

Comment:
consider switching to a 'switch' statement, instead of a bunch of if/else... this particular scenario is perfectly suited to a switch statement, and would perform slightly better, as it would only require 1 evaluation instead of up to 8